src/encryption.ts

```diff
@@ -0,0 +1,84 @@
+interface GeneralRecipient {
+  encrypted_key?: string;
+  header?: Object;
+}
+
+interface GeneralJWE {
+  aad?: string;
+  ciphertext: string;
+  iv?: string;
+  protected?: string;
+  recipients: Array<GeneralRecipient>;
+  tag?: string;
+  unprotected?: Object;
+}
+
+interface DagRecipient {
+  encrypted_key?: Buffer;
```

Comment:
Do you mean that they could have a shared interface just for the header type?

Reply:
I'm wondering if the `encrypted_key` type could be the same in both interfaces to avoid possible confusions between `Buffer` and `string`?
If so we could simply have `type DagRecipient = GeneralRecipient` or just removing `DagRecipient` in favor of `GeneralRecipient`.

Reply:
Ah, I see. The idea here is that it should be decoded by this library such that it makes it easier for the decryption function to consume more directly: https://www.stablelib.com/classes/_xchacha20poly1305_xchacha20poly1305_.xchacha20poly1305.html

Reply:
👍 I see, thanks!
I guess it depends if this interface is used only at the edge (right before the decryption call) or it can be passed around?
I'm thinking about systems where for example there could be chained JSON-RPC calls (such app -> iframe -> HTTP server), where it would unnecessary to (de)serialize the key?

Reply:
Yeah, could depend a lot on the setting. I don't think we should overoptimize for that particular use case before we know more exactly what things will be passed where!